### PR TITLE
security, server: fix running local test

### LINKED ISSUE #12711: *: `cargo test` cannot compile
### What version of TiKV are you using?
Master

### What operating system and CPU are you using?
Linux

### Steps to reproduce
Run test for a subcrate requires `security` but not `backup-stream`.

### What did you expect?
Test can run successfully.

### What did happened?
Compile failed: 
```console
use tonic::transport::{channel::ClientTlsConfig, Certificate, Identity};
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^ no `ClientTlsConfig` in `transport::channel`
```

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: Yu Juncen <yujuncen@pingcap.com>

### What is changed and how it works?

Issue Number: Close #12711 

What's Changed:
Make `tonic` in `security` crate an optional requirement.

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 
Now we can run `cargo test` for raftstore. And there is no need for compiling tonic.

image

- No code

### Release note 

```release-note
none
```

### CODE REVIEW

**Review by 3pointer — APPROVED** [review]

**Comment by tonyxuqqi** [comment]
LGTM

**Review by kennytm — APPROVED** [review]

**Comment by YuJuncen** [comment]
`[2022-06-01T07:27:32.008Z] Setting status of b9d767e3b8119b4d24aef60e80521424989db1f1 to PENDING with url https://ci.pingcap.net/job/tikv_ghpr_build/15145/display/redirect and message: 'Jenkins job is running.'`
🤔
/rebuild

File: components/backup-stream/tests/mod.rs [hunk 1]
```diff
@@ -628,26 +628,16 @@ mod test {
...
-        assert_eq!(safepoints.len(), 4, "{:?}", safepoints);
+
         assert!(
-            safepoints
-                .iter()
-                .take(3)
-                // They are choosing the lock safepoint, it must greater than the global checkpoint.
-                .all(|sp| { sp.safepoint.into_inner() >= checkpoint }),
+            safepoints.iter().any(|sp| {
+                sp.serivce.contains(&format!("{}", victim))
```
Comment (3pointer) [inline components/backup-stream/tests/mod.rs:634]: maybe use `partition_in_place` by `sp.serivce.contains(&format!("{}", victim)` and check tow parts is more precise.

**Review by 3pointer — APPROVED** [review]
rest LGTM

File: components/backup-stream/tests/mod.rs (hunk 1 above)
Comment (YuJuncen) [reply components/backup-stream/tests/mod.rs:634]: The problem is the flush procedure is asynchronous: which means it sometime won't take place. Checking them may get failed and is out of the scope of `fatal_error`.

**Pull request merged**